File: JS/Tree/429. N-ary Tree Level Order Traversal.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */

/**
 * @param {Node} root
 * @return {number[][]}
 */
var levelOrder = function(root) {
    const result = [];
    const stack = [root];
    let count = 1;
    let current = 0;
    let depth = 0;
    while(stack.length) {
        const node = stack.shift();
        if (node) {
            if (current === 0) {
                result.push([]);
            }
            current++;
            result[depth].push(node.val);
            node.children.forEach(i => stack.push(i));
            if (current === count) {
                current = 0;
                count = stack.length;
                depth++;
            }
        }
    }
    return result;
};

File: Golang/Tree/429.n-ary-tree-level-order-traversal.go
package main

type Node struct {
	Val      int
	Children []*Node
}

func levelOrder(root *Node) [][]int {
	ret := [][]int{}
	if root != nil {
		stack := []*Node{root}
		for len(stack) > 0 {
			layer := []int{}
			children := []*Node{}
			for i := 0; i < len(stack); i++ {
				node := stack[i]
				layer = append(layer, node.Val)
				children = append(children, node.Children...)
			}
			ret = append(ret, layer)
			stack = children
		}
	}
	return ret
}
